### PR TITLE
chore: DH-15527 Move ACL Editor Code to Community

### PR DESCRIPTION
**Pull request by bmingles**
Moved ACL Editor util code from Enterprise to Community. This should not introduce any functional changes.

Supports #1469 and DH-15527

### CODE REVIEW

File: packages/utils/package.json [hunk 1]
```diff
@@ -27,5 +27,8 @@
   "sideEffects": false,
   "publishConfig": {
     "access": "public"
+  },
+  "dependencies": {
+    "@adobe/react-spectrum": "^3.29.0"
```
Comment (mofojed) [inline packages/utils/package.json:32]: This seems like a pretty large dependency to add to `@deephaven/utils` which currently has no other dependencies. There's basically two things added to utils that require this:
1. `SpectrumUtils`: That's only used within `@deephaven/components` and `SelectionUtils`, which is also used in `react-hooks`. Should probably move it in there.
2. `WindowedListData`: Used within jsapi-components, jsapi-utils, react-hooks. jsapi-components depends on jsapi-utils which depends on react-hooks, so move `WindowedListData`, so I think `react-hooks` would be a more appropriate place for that.

`@deephaven/utils` is really just for vanilla utility methods, and shouldn't have any large dependencies.

**Review by mofojed — CHANGES_REQUESTED** [review]
Only issue I think is the dependency of `utils` on `@adobe/react-spectrum`. That dependency should be lifted up to `react-hooks` and out of `utils`.

File: packages/utils/package.json (hunk 1 above)
Comment (bmingles) [reply packages/utils/package.json]: Spectrum dependency removed from @deephaven/utils

File: packages/jsapi-utils/src/ViewportDataUtils.ts [hunk 2]
```diff
@@ -2,7 +2,8 @@ import type { Key } from 'react';
 import clamp from 'lodash.clamp';
 import type { Column, Row, Table, TreeTable } from '@deephaven/jsapi-types';
 import Log from '@deephaven/log';
-import { KeyedItem, WindowedListData } from '@deephaven/utils';
+import type { WindowedListData } from '@deephaven/react-hooks';
```
Comment (mofojed) [inline packages/jsapi-utils/src/ViewportDataUtils.ts:5]: jsapi-utils depending on react-hooks feels wrong as well. I think moving the dependencies here to jsapi-components instead (those are where React components that use the jsapi live).

**Review by mofojed — CHANGES_REQUESTED** [review]

File: packages/jsapi-utils/src/ViewportDataUtils.ts (hunk 2 above)
Comment (bmingles) [reply packages/jsapi-utils/src/ViewportDataUtils.ts:5]: @mofojed can you elaborate on this one? jsapi-components depends on jsapi-utils. Which dependencies were you thinking of moving to jsapi-components?

Comment (bmingles) [reply packages/jsapi-utils/src/ViewportDataUtils.ts]: I was able to remove the react-hooks dependency by reconfiguring some types

**Review by mofojed — APPROVED** [review]

**Pull request merged**